File: ManyInOneFront/src/app/shared/interfaces/player.ts
export interface IPlayerResponse {
    succeed: boolean;
    message: any;
    errors: any;
    result: IPlayer;
}

export interface IPlayer {
    tag: string;
    name: string;
    townHallLevel: number;
    townHallWeaponLevel: number;
    expLevel: number;
    trophies: number;
    bestTrophies: number;
    warStars: number;
    attackWins: number;
    defenseWins: number;
    builderHallLevel: number;
    builderBaseTrophies: number;
    bestBuilderBaseTrophies: number;
    role: string;
    warPreference: string;
    donations: number;
    donationsReceived: number;
    clanCapitalContributions: number;
    clan: IClan;
    league: ILeague;
    builderBaseLeague: IBuilderBaseLeague;
    legendStatistics: ILegendStatistics;
    achievements: IAchievement[];
    playerHouse: IPlayerHouse;
    labels: ILabel[];
    troops: ITroop[];
    heroes: IHero[];
    heroEquipment: IHeroEquipment[];
    spells: ISpell[];
}
export interface ILabel {
    id: number;
    name: string;
    iconUrls: IIconUrls;
}
export interface IClan {
    tag: string;
    name: string;
    clanLevel: number;
    badgeUrls: IBadgeUrls;
}

export interface IBadgeUrls {
    small: string;
    large: string;
    medium: string;
}

export interface ILeague {
    id: number;
    name: string;
    iconUrls: IIconUrls;
}

export interface IIconUrls {
    small: string;
    tiny: string;
    medium: string;
}

export interface IBuilderBaseLeague {
    id: number;
    name: string;
}

export interface ILegendStatistics {
    legendTrophies: number;
    previousSeason: IPreviousSeason;
    bestSeason: IBestSeason;
    currentSeason: IICurrentSeason;
}

export interface IPreviousSeason {
    id: string;
    rank: number;
    trophies: number;
}

export interface IBestSeason {
    id: string;
    rank: number;
    trophies: number;
}

export interface IICurrentSeason {
    rank: number;
    trophies: number;
}

export interface IAchievement {
    name: string;
    stars: number;
    value: number;
    target: number;
    info: string;
    completionInfo?: string;
    village: string;
}

export interface IPlayerHouse {
    elements: IElement[];
}

export interface IElement {
    id: number;
    type: string;
}

export interface ITroop {
    name: string;
    level: number;
    maxLevel: number;
    village: string;
    superTroopIsActive: boolean;
}

export interface IHero {
    name: string;
    level: number;
    maxLevel: number;
    equipment?: IEquipment[];
    village: string;
}

export interface IEquipment {
    name: string;
    level: number;
    maxLevel: number;
    village: string;
}

export interface IHeroEquipment {
    name: string;
    level: number;
    maxLevel: number;
    village: string;
}

export interface ISpell {
    name: string;
    level: number;
    maxLevel: number;
    village: string;
}

File: ManyInOneAPI/Models/Clasher/Player.cs
namespace ManyInOneAPI.Models.Clasher
{
    public class Player
    {
        public string? tag { get; set; }
        public string? name { get; set; }
        public int townHallLevel { get; set; }
        public int townHallWeaponLevel { get; set; }
        public int expLevel { get; set; }
        public int trophies { get; set; }
        public int bestTrophies { get; set; }
        public int warStars { get; set; }
        public int attackWins { get; set; }
        public int defenseWins { get; set; }
        public int builderHallLevel { get; set; }
        public int builderBaseTrophies { get; set; }
        public int bestBuilderBaseTrophies { get; set; }
        public string? role { get; set; }
        public string? warPreference { get; set; }
        public int donations { get; set; }
        public int donationsReceived { get; set; }
        public int clanCapitalContributions { get; set; }
        public Clan? clan { get; set; }
        public League? league { get; set; }
        public BuilderBaseLeague? builderBaseLeague { get; set; }
        public LegendStatistics? legendStatistics { get; set; }
        public List<Achievement>? achievements { get; set; }
        public PlayerHouse? playerHouse { get; set; }
        public List<Label>? labels { get; set; }
        public List<Troop>? troops { get; set;}
        public List<Hero>? heroes { get; set; }
        public List<HeroEquipment>? heroEquipment {  get; set; }
        public List<Spell>? spells { get; set; }
    }

    public class Clan
    {
        public string? tag { get; set; }
        public string? name { get; set; }
        public int clanLevel { get; set; }
        public BadgeUrls? badgeUrls { get; set; }
        public class BadgeUrls
        {
            public string? small { get; set; }
            public string? large { get; set; }
            public string? medium { get; set; }
        }
    }
    public class League
    {
        public int id { get; set; }
        public string? name { get; set; }
        public IconUrls? iconUrls { get; set; }
    }
    public class IconUrls
    {
        public string? small { get; set; }
        public string? tiny { get; set; }
        public string? medium { get; set; }
    }
    public class BuilderBaseLeague
    {
        public int id { get; set; }
        public string? name { get; set; }
    }
    public class LegendStatistics
    {
        public int legendTrophies { get; set; }
        public PreviousSeason? previousSeason { get; set; }
        public BestSeason? bestSeason { get; set; }
        public CurrentSeason? currentSeason { get; set; }

        public class PreviousSeason
        {
            public string? id { get; set; }
            public int rank { get; set; }
            public int trophies { get; set; }
        }
        public class BestSeason
        {
            public string? id { get; set; }
            public int rank { get; set; }
            public int trophies { get; set; }
        }

        public class CurrentSeason
        {
            public int rank { get; set; }
            public int trophies { get; set; }
        }
    }
    public class Achievement
    {
        public string? name { get; set; }
        public int stars { get; set; }
        public int value { get; set; }
        public int target { get; set; }
        public string? info { get; set; }
        public string? completionInfo { get; set; }
        public string? village { get; set; }
    }
    public class PlayerHouse
    {
        public List<Element>? elements { get; set;}

        public class Element 
        {
            public int id { get; set; }
            public string? type { get; set; }
        }
    }
    public class Label
    {
        public int id { get; set; }
        public string? name { get; set; }
        public IconUrls? iconUrls { get; set; }
        public class IconUrls
        {
            public string? small { get; set; }
            public string? medium { get; set; }
        }
    }
    public class Troop
    {
        public string? name { get; set; }
        public int level { get; set; }
        public int maxLevel { get; set; }
        public string? village { get; set; }
        public bool superTroopIsActive { get; set; }
    }
    public class Hero
    {
        public string? name { get; set; }
        public int level { get; set; }
        public int maxLevel { get; set; }
        public List<Equipment>? equipment { get; set; }
        public string? village { get; set; }
    }
    public class Equipment
    {
        public string? name { get; set; }
        public int level { get; set; }
        public int maxLevel { get; set; }
        public string? village { get; set; }
    }
    public class HeroEquipment
    {
        public string? name { get; set; }
        public int level { get; set; }
        public int maxLevel { get; set; }
        public string? village { get; set; }
    }
    public class Spell
    {
        public string? name { get; set; }
        public int level { get; set; }
        public int maxLevel { get; set; }
        public string? village { get; set; }
    }
}
